File: app/src/main/java/gardezi/io/amiibofinder/AmiiboApplication.java
package gardezi.io.amiibofinder;

import android.app.Activity;
import android.app.Application;

import javax.inject.Inject;

import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;
import gardezi.io.amiibofinder.dagger.injector.AppInjector;

public class AmiiboApplication extends Application implements HasActivityInjector {

    private static AmiiboApplication instance;

    @Inject
    DispatchingAndroidInjector<Activity> dispatchingAndroidInjector;

    public static AmiiboApplication getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        AppInjector.init(this);
    }

    @Override
    public DispatchingAndroidInjector<Activity> activityInjector() {
        return dispatchingAndroidInjector;
    }

}

File: app/src/main/java/gardezi/io/amiibofinder/AmiiboApplication.kt
package gardezi.io.amiibofinder

import android.app.Activity
import android.app.Application
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import gardezi.io.amiibofinder.dagger.AppInjector
import javax.inject.Inject

class AmiiboApplication : Application(), HasActivityInjector {

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>

    override fun onCreate() {
        super.onCreate()
        instance = this
        AppInjector.init(this)
    }

    override fun activityInjector(): DispatchingAndroidInjector<Activity>? {
        return dispatchingAndroidInjector
    }

    companion object {
        var instance: AmiiboApplication? = null
            private set
    }

}
